### DISCUSSION TITLE
Having "Wants ... " Annotation variants?

### DISCUSSION THREAD

**Discussion by laeubi** [osgi/osgi#568] [Ideas]
Things like `RequireConfigurationAdmin`, `RequireServiceComponentRuntime`, ... are already a good way to express a strict requirement.

In some cases it might be good to be not that strict, but more express that the bundle wants something if available, to help the resolver e.g. see https://docs.osgi.org/specification/osgi.core/8.0.0/service.resolver.html#d0e50371 it says:

>  The resolver _must_ find a solution that includes at least the initial mandatory resources and _should_ include the optional resources 

So my idea would be to have a corresponding set of `Wants...` annotations that simply set the resolution to optional, so I can tell the resolver I want it if it can be found but can also work (maybe with limited features) without it.

**Comment by stbischof** [comment]
Something like an optional requiroment?

We have this for optionale scr compinents.

**Reply by laeubi** [reply]
Yes, like optional components, a bundle _might_ offer additional function when more extenders is available but can also work _without_ it.
